### ISSUE TITLE
Strange loss values

### ISSUE THREAD

**Issue by NguyenThanhTan** [zisianw/FaceBoxes.PyTorch#44]
While training the model, training loss looks strange 

> Epoch:1 || epochiter: 0/316|| Total iter 316 || L: 5.69 C: 0.17 ||Btime: 26.588 s. ||LR: 0.001085
> Epoch:1 || epochiter: 1/316|| Total iter 317 || L: 6.12 C: 0.23 ||Btime: 1.360 s. ||LR: 0.001085
> Epoch:1 || epochiter: 2/316|| Total iter 318 || L: 6.23 C: 0.44 ||Btime: 3.178 s. ||LR: 0.001086
> Epoch:1 || epochiter: 3/316|| Total iter 319 || L: 6.27 C: 0.35 ||Btime: 1.331 s. ||LR: 0.001086
> Epoch:1 || epochiter: 4/316|| Total iter 320 || L: 5.74 C: 0.19 ||Btime: 1.145 s. ||LR: 0.001086
> Epoch:1 || epochiter: 5/316|| Total iter 321 || L: 6.13 C: 0.29 ||Btime: 2.172 s. ||LR: 0.001086
> Epoch:1 || epochiter: 6/316|| Total iter 322 || L: 5.08 C: 0.20 ||Btime: 1.199 s. ||LR: 0.001087
> Epoch:1 || epochiter: 7/316|| Total iter 323 || L: 5.68 C: 0.24 ||Btime: 0.900 s. ||LR: 0.001087
> Epoch:1 || epochiter: 8/316|| Total iter 324 || L: 5.54 C: 0.62 ||Btime: 1.470 s. ||LR: 0.001087
> Epoch:1 || epochiter: 9/316|| Total iter 325 || L: 5.91 C: 0.31 ||Btime: 0.917 s. ||LR: 0.001088

while validation loss is like this:

> Epoch:1 || L: 4.5863 C: 45.4624||Batch time: 14.5824 sec. ||
> Epoch:1 || L: 4.7383 C: 44.9612||Batch time: 0.9316 sec. ||
> Epoch:1 || L: 4.6935 C: 44.3159||Batch time: 1.3099 sec. ||
> Epoch:1 || L: 4.4057 C: 45.0097||Batch time: 0.8248 sec. ||
> Epoch:1 || L: 4.2893 C: 45.4343||Batch time: 0.7287 sec. ||

I know that the model is kind of collapsing so I run inference on an image to see the behavior of the model. I observe that it predicts almost all anchors are faces (thousands of boxes on one image) on images from either train or validation dataset. This complies to the validation loss above but my question is why the classification loss of training is so low while the model predicts thousands of boxes for an image. Below is the inference run on a training image:
![13_Interview_Interview_On_Location_13_870](https://user-images.githubusercontent.com/17760790/56630490-bed2b000-667b-11e9-8e2a-7a1ec8fff357.jpg)

**Comment by quancq** [comment]
Is training data imbalance (very large number positive anchor and very low number negative anchor)?

**Comment by NguyenThanhTan** [comment]
No, my training data is wider face dataset. But that is not what I'm confused.
The strange thing is the training loss, specifically classification loss, is so low while the model is collapsed (and validation classification loss is high, but this is expected).
